### ISSUE TITLE
RadioMessages send function

### ISSUE THREAD

**Issue by JakobGartner** [openETCS/modeling#445]
In BalisesLib, add radio-message send function as discussed with @aprobst

**Comment by aprobst** [comment]
@JakobGartner, the operator TM_lib_internal::SEND_MergeRadioPackets is missing in TrackMessages.etp. Is this only a temporary issue? However I would like to point out that this operator is essential for the behavior of the RBC.

**Comment by JakobGartner** [comment]
This operator was a temporary solution.
The problem is that it is not generic enough ( it needs to be changed for each combination of packets) 

If we want to be closer to the SRS also on message level, a more modular solution is needed which will be provided.  I expect to have it ready before monday morning.

**Comment by JakobGartner** [comment]
As a temporary solution, the function has been put back into the project: TM_old_versions::SEND_MergeRadioPackets

**Comment by BerndHekele** [comment]
anks.
I will take the solution as a basis for the design as agreed on during this weeks meeting.

**Comment by JakobGartner** [comment]
This is a trackside operator.

I will send you the pattern for the Trainside operator later today ( in Telco now)

**Comment by aprobst** [comment]
TM_old_versions::SEND_MergeRadioPackets is not available in TrackMessages.etp. From my point of view, this old operator fulfilled all my requirements because it made it possible to create manually compressed messages with self defined factory operators (see Package Messaging_pkg::RadioTrackTrainFactory_Pkg). However as you already said, this operator seems not to be generic engough to satisfy the SRS. Nevertheless, I am not able to continue and to simulate due to the above named problem.

Where can I then find the new radio operator?

In conclusion I want also to refer to issue #460 that contains a possible recommendation for triggering radio based Track-to-Train messages.

@BaseliyosJacob

**Comment by mairamou** [comment]
TM_old_versions::SEND_MergeRadioPackets has several issues (it was a first try) that make it hard to use in a universal way.

As the new operator is still in my backlog, I have “renovated” this one and pushed it anew.

Please note that the basic data types have been changed in order to be fully SRS compliant. 


Please also note that the new write function may contain bugs as we are doing integration testing with the full track model.

**Comment by JakobGartner** [comment]
done, now there is a package TM_radio_messages in TrackMessages and a corresponding package RadioLib in BaliseLib

**Comment by BaseliyosJacob** [comment]
- [ ] Analysis
- [ ] Modelling
- [ ] Documentation
- [ ] Handover to V&V
- [ ] Approval from V&V

**Comment by JakobGartner** [comment]
done for TrackToTrain , Train to Track is for next iteration
